mp/src/game/shared/momentum/mom_stickybomb.cpp

```diff
@@ -0,0 +1,387 @@
+}
+
+#ifdef CLIENT_DLL
+
+void CMomStickybomb::OnDataChanged(DataUpdateType_t type)
+{
+    BaseClass::OnDataChanged(type);
+
+    if (type == DATA_UPDATE_CREATED)
+    {
+        m_flCreationTime = gpGlobals->curtime;
+        ParticleProp()->Create("stickybombtrail_red", PATTACH_ABSORIGIN_FOLLOW);
+        m_bPulsed = false;
+
+        CMomentumStickybombLauncher *pLauncher = dynamic_cast<CMomentumStickybombLauncher *>(m_hLauncher.Get());
+
+        if (pLauncher)
+        {
+            pLauncher->AddStickybomb(this);
+        }
+
+        // Now stick our initial velocity into the interpolation history
+        CInterpolatedVar<Vector> &interpolator = GetOriginInterpolator();
+        interpolator.ClearHistory();
+
+        CInterpolatedVar<QAngle> &rotInterpolator = GetRotationInterpolator();
+        rotInterpolator.ClearHistory();
+
+        float changeTime = GetLastChangeTime(LATCH_SIMULATION_VAR);
+
+        // Add a sample 1 second back.
+        Vector vCurOrigin = GetLocalOrigin() - m_vInitialVelocity;
+        interpolator.AddToHead(changeTime - 1.0f, &vCurOrigin, false);
+
+        QAngle vCurAngles = GetLocalAngles();
+        rotInterpolator.AddToHead(changeTime - 1.0f, &vCurAngles, false);
+
+        // Add the current sample.
+        vCurOrigin = GetLocalOrigin();
+        interpolator.AddToHead(changeTime, &vCurOrigin, false);
+
+        rotInterpolator.AddToHead(changeTime, &vCurAngles, false);
+    }
+}
+
+int CMomStickybomb::DrawModel(int flags)
+{
+    // Don't draw the stickybomb during the time defined by the cvar.
+    if (gpGlobals->curtime - m_flSpawnTime < mom_sj_stickybomb_drawdelay.GetFloat())
+    {
+        return 0;
+    }
+
+    return BaseClass::DrawModel(flags);
+}
+
+void CMomStickybomb::Spawn()
+{
+    m_flSpawnTime = gpGlobals->curtime;
+    BaseClass::Spawn();
+}
+
+void CMomStickybomb::Simulate()
+{
+    if (!m_bPulsed)
+    {
+        if ((gpGlobals->curtime - m_flCreationTime) >= 0.8f)
+        {
+            ParticleProp()->Create("stickybomb_pulse_red", PATTACH_ABSORIGIN_FOLLOW);
+            m_bPulsed = true;
+        }
+    }
+    BaseClass::Simulate();
+}
+
+#else
+
+void CMomStickybomb::Spawn()
+{
+    SetModel(MOM_STICKYBOMB_MODEL);
+    BaseClass::Spawn();
+
+    UseClientSideAnimation();
+    SetCollisionGroup(COLLISION_GROUP_PROJECTILE);
+    SetSolidFlags(FSOLID_NOT_STANDABLE);
+    SetMoveType(MOVETYPE_FLYGRAVITY, MOVECOLLIDE_FLY_BOUNCE);
+    SetSolid(SOLID_BBOX);
+    AddEffects(EF_NOSHADOW);
+    SetSize(Vector(-2, -2, -2), Vector(2, 2, 2));
+    AddFlag(FL_GRENADE);
+
+    VPhysicsInitNormal(SOLID_BBOX, 0, false);
+
+    m_bTouched = false;
+    m_flCreationTime = gpGlobals->curtime;
+    m_takedamage = DAMAGE_NO;
+    SetGravity(MOM_STICKYBOMB_GRAVITY);
+    SetFriction(MOM_STICKYBOMB_FRICTION);
+    SetElasticity(MOM_STICKYBOMB_ELASTICITY);
+
+    SetTouch(&CMomStickybomb::StickybombTouch);
+    SetThink(&CMomStickybomb::StickybombThink);
+    SetNextThink(gpGlobals->curtime + 0.2);
```

Comment:
If you are removing your think/touch functions, don't forget to remove these as well.